### PR TITLE
docs(swagger): setup swagger docs

### PR DESCRIPTION
**Pull request by aleem1314**
closes: #161

### CODE REVIEW

**Comment by amaurymartiny** [comment]
As pointed out in https://github.com/regen-network/regen-ledger/issues/161#issuecomment-726075064, don't we need to add grpc-gateway annotations before generating swagger docs?

**Comment by anilCSE** [comment]
> As pointed out in [#161 (comment)](https://github.com/regen-network/regen-ledger/issues/161#issuecomment-726075064), don't we need to add grpc-gateway annotations before generating swagger docs?

I think that can be addressed in a separate PR as well. The config file here includes endpoints from regen modules too, it will be just regenerating the swagger docs when we add those annotations.

**Comment by anilCSE** [comment]
@aleem1413 since we have all the proto files from cosmos, can we include them in the script as well? Let's avoid copying/update cosmos' swagger file manually

**Comment by aleem1413** [comment]
> @aleem1413 since we have all the proto files from cosmos, can we include them in the script directly to generate cosmos' swagger? Let's avoid copying/update cosmos' swagger file manually.

@anilCSE We don't have cosmos proto files in the regen-ledger.

File: scripts/protoc-swagger-gen.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "$query_file" \
+    --swagger_out ./tmp-swagger-gen \
+    --swagger_opt logtostderr=true --swagger_opt fqn_for_swagger_name=true --swagger_opt simple_operation_ids=true
+  fi
+done
+
+
+# download Cosmos SDK swagger doc
+rm -f ./client/docs/swagger.yaml
+wget https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.40.0/client/docs/swagger-ui/swagger.yaml -P ./client/docs
```
Comment (anilCSE) [inline scripts/protoc-swagger-gen.sh:24]: @aleem1314 can you update this URL to 0.41? And regenrate the docs

File: scripts/protoc-swagger-gen.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+# combine swagger files
+# uses nodejs package `swagger-combine`.
+# all the individual swagger files need to be configured in `config.json` for merging
+swagger-combine ./client/docs/config.json -o ./client/docs/swagger-ui/swagger.yaml -f yaml --continueOnConflictingPaths true --includeDefinitions true
+
+# clean swagger files
+rm -rf ./tmp-swagger-gen
```
Comment (anilCSE) [inline scripts/protoc-swagger-gen.sh:32]:
```suggestion
rm -rf ./tmp-swagger-gen

```

File: client/docs/config.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            "url": "./tmp-swagger-gen/regen/group/v1alpha1/query.swagger.json"
+        },
+        {
+            "url": "./client/docs/swagger.yaml",
+            "dereference": {
+                "circular": "ignore"
+            }
+        }
+    ]
+}
```
Comment (anilCSE) [inline client/docs/config.json:25]:
```suggestion
}

```

**Review by anilCSE — CHANGES_REQUESTED** [review]

**Review by anilCSE — APPROVED** [review]

File: client/docs/config.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "title": "Regen Ledger - REST and gRPC Gateway docs",
```
Comment (ryanchristo) [inline client/docs/config.json:4]:
```suggestion
        "title": "Regen Ledger - REST and gRPC Gateway Documentation",
```

**Review by ryanchristo — COMMENTED** [review]
Looking good. A few comments and suggestions. Also, can we update the "Gaia REST" tag for the `node_info` endpoint. [Here](https://www.npmjs.com/package/swagger-combine#renaming-tags) is the `swagger-combine` documentation to do so.

File: scripts/protoc-swagger-gen.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+echo "SDK version ${SDK_VERSION}"
+wget "https://raw.githubusercontent.com/cosmos/cosmos-sdk/${SDK_VERSION}/client/docs/swagger-ui/swagger.yaml" -P ./client/docs
+
+# combine swagger files
+# uses nodejs package `swagger-combine`.
+# all the individual swagger files need to be configured in `config.json` for merging
+swagger-combine ./client/docs/config.json -o ./client/docs/swagger-ui/swagger.yaml -f yaml --continueOnConflictingPaths true --includeDefinitions true
+
+# clean swagger files
+rm -rf ./tmp-swagger-gen
```
[the remark this answers is not in the document]
Comment (aleem1314) [reply scripts/protoc-swagger-gen.sh:34]: > Or maybe we download this file into the tmp-swagger-gen directory?

Make sense.

**Review by ryanchristo — APPROVED** [review]
LGTM

**Pull request merged**